steward/src/cellars/adaptors/balancer_pool.rs

```diff
@@ -0,0 +1,174 @@
+use ethers::{abi::AbiEncode, types::Bytes};
+use steward_abi::balancer_pool_adaptor_v1::{
+    BalancerPoolAdaptorV1Calls, ExitPoolRequest, SwapData,
+};
+use steward_proto::steward::balancer_pool_adaptor_v1;
+
+use crate::{
+    error::Error,
+    utils::{sp_call_error, sp_call_parse_address, sp_call_parse_h256, string_to_u256},
+};
+
+pub(crate) fn balancer_pool_adaptor_v1_calls(
+    params: steward_proto::steward::BalancerPoolAdaptorV1Calls,
+) -> Result<Vec<Bytes>, Error> {
+    let mut calls = Vec::new();
+    for c in params.calls {
+        let function = c
+            .function
+            .ok_or_else(|| sp_call_error("function cannot be empty".to_string()))?;
+
+        match function {
+            balancer_pool_adaptor_v1::Function::RevokeApproval(p) => {
+                let call = steward_abi::balancer_pool_adaptor_v1::RevokeApprovalCall {
+                    asset: sp_call_parse_address(p.asset)?,
+                    spender: sp_call_parse_address(p.spender)?,
+                };
+                calls.push(
+                    BalancerPoolAdaptorV1Calls::RevokeApproval(call)
+                        .encode()
+                        .into(),
+                )
+            }
+            balancer_pool_adaptor_v1::Function::JoinPool(p) => {
+                if !p.swaps_before_join.is_empty()
+                    && p.swaps_before_join.iter().any(|s| s.kind == 0)
+                {
+                    return Err(sp_call_error("invalid swap kind".to_string()));
+                }
+
+                let swaps_before_join = p
+                    .swaps_before_join
+                    .into_iter()
+                    .map(|s| {
+                        Ok(steward_abi::balancer_pool_adaptor_v1::SingleSwap {
+                            pool_id: sp_call_parse_h256(s.pool_id)?.into(),
```

Comment:
This is the first time we've parsed something as an `h256`, is there a reason we're using the hash type here rather than some raw bytes? We're calling `into` so it's not the type the ABI is expecting, right? What is the ABI type?

Reply:
the ABI calls for a bytes32, which is bound to in rust as a [u8; 32]. This function parses the string as its 32 byte hex representation, and `into()` casts it to the array infallibly.

Reply:
So I guess we're using h256 for convenience since it's backed by a `[u8; 32]` and has some conversion functions already built in? Just odd because it really has nothing to do with a hash type.

Reply:
hmm yeah. actually we should probably check if they are getting the pool_id as hex and then we could just change the param to a string and this parsing would make more sense.